Bundle fixture classes

twojtylak opened the pull request.

Commit:
Added basic ProductBundleFixture classes

Commit:
Fixtures for product bundles

Commit:
Implemented ProductBundle Fixtures Classes & added them as service

Commit:
Adjusted fixture: Define football product bundle

Commit:
Use ProductBundleCreator in ProductBundleExampleFactory

Commit:
Added cascade: ['persist'] for ProductBundle.orm.yml to persist slot entity without calling persist() explicitly

Commit:
Create ProductBundleSlots via productBundleCreator

Commit:
Adjusted fixture structure for slots

Commit:
resolve slotOptions

Commit:
fixed code styles

Commit:
Added Readme entry for Fixtures

Commit:
Fixed styles in README.md (removed additional '#')

Commit:
Fixed code styles

solutionDrive-Alt commented on src/Fixture/Factory/ProductBundleExampleFactory.php:
```diff
@@ -0,0 +1,111 @@
+            Assert::methodExists($slotOptions, $setter, sprintf('Setter %s for ProductBundleSlotOptions is not defined', $setter));
+
+            if (method_exists($slotOptions, $setter = 'set' . ucfirst($optionName))) {
+                $slotOptions->$setter($optionValue);
```
Just as a note, there is also the possibility to add additional options with the 
[ProductBundleSlotOptionsInterface::addOptionMethod](https://github.com/solutionDrive/SyliusProductBundlesPlugin/blob/84d57030be3646700c18a98d4c3d1df413625c48/src/Service/Options/OptionsInterface.php#L14) in an else-branch if that is a thing that is needed.

mattagohni commented on src/Fixture/Factory/ProductBundleExampleFactory.php:
```diff
@@ -0,0 +1,111 @@
+            Assert::methodExists($slotOptions, $setter, sprintf('Setter %s for ProductBundleSlotOptions is not defined', $setter));
+
+            if (method_exists($slotOptions, $setter = 'set' . ucfirst($optionName))) {
+                $slotOptions->$setter($optionValue);
```
There is also the possibility to add additional options with the [ProductBundleSlotOptionsInterface::addOptionMethod](https://github.com/solutionDrive/SyliusProductBundlesPlugin/blob/84d57030be3646700c18a98d4c3d1df413625c48/src/Service/Options/OptionsInterface.php#L14) if there is a use case for this. But maybe this is more important in the project, that uses this plugin

twojtylak replied:
yeah thats better. I'll change this tomorrow. No idea why I didn't see this function.

twojtylak replied:
Hmm I think for now we don't need additional options. There are not used at the moment anyway. Maybe the naming is a little bit confusing (they are saved in a variable additionalOptions) ...

mattagohni replied:
@twojtylak yeah, that's correct. The idea behind that is, that you have fixed options for a ProductBundleSlotOptions-Object (position/isPresentationSlot) and the possibility to add custom options that can influence the BundleSlotCreation. (e.g. when someone implements his own BundleSlot-Class).

mattagohni commented on src/Resources/config/services.yml:
```diff
@@ -19,3 +19,17 @@ services:
             - "@solutiondrive.factory.product_bundle"
             - "@solutiondrive.factory.product_bundle_slot"
+
+    solutiondrive.product_bundles.fixture.example_factory.product_bundle:
```
Should the example-factories be defined in the 'main'-services.yml? Maybe you don't really want these services in your prod-environment...

twojtylak replied:
Neither sylius nor any plugin I've seen makes a difference between prod and dev-services. So I don't think we should..

mattagohni replied:
ok, was only a thought... :-)

mattagohni commented:
@twojtylak  can you please rebase? we have conflicts.

Commit:
Adjusted DB-Schema: Delete connected DB-Data on deletion of ProductBundles

twojtylak commented:
I rebased the branch right now. There were some  integrity constraint violations when you tried to delete a ProductBundle with assigned Slots. So I added the onDelete cascade

Commit:
Trigger
